utils/communication_report.py

```diff
@@ -0,0 +1,133 @@
+'''
+This script generates a CSV file of all Communication resources in the system.
+It's intended to be run from the /dev directory of the deployment that you want to 
+report from; it requires no modification to be run from that location.
+It accepts no arguments.
+'''
+
+import subprocess
+import json
+import requests
+import csv
+import urllib.parse
+
+base_url = "http://fhir-internal:8080/fhir"
+
+def get_fhir_resource(url):
+
+    print(f"get_fhir_resource({url}), just entered.")
+
+    command = f"docker-compose exec femr curl -X GET {url}"
+    result = subprocess.run(command, shell=True, capture_output=True, text=True)
```

Comment:
it'd be less error-prone to run this script inside a container, with `requests`. 

using `docker-compose run` can save a few `docker cp` steps:
```
docker-compose run --volume=${PWD}/utils:/opt/utils femr \
    python3 /opt/utils/communication_report.py
```

Reply:
You highlighted lines for a subprocess call, but what you wrote seems to be referring to running the entire script. Perhaps I should add that dc call to the comments at the top of the file?

Reply:
>You highlighted lines for a subprocess call, but what you wrote seems to be referring to running the entire script. 

Invoking the entire script via `docker-compose run` (vs on the VM) would be to replace the `subprocess` with `requests` calls

Having the script make HTTP calls directly (vs curl) would require turning this script inside-out, ie invoking the script via docker-compose vs having the script invoke docker-compose. This would make the script less dependent on docker and able to work on any FHIR HTTP endpoint.


>Perhaps I should add that dc call to the comments at the top of the file?

Yup! that would be a great place to document how to run it

Reply:
I've just added a comment describing how to run it; merging now.